include/zephyr/drivers/sent/sent.h

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright 2025 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * @file
+ * @brief Single Edge Nibble Transmission (SENT) driver API.
+ */
+
+#ifndef ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+#define ZEPHYR_INCLUDE_DRIVERS_SENT_H_
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief SENT Interface
+ * @defgroup sent_interface SENT Interface
+ * @ingroup io_interfaces
+ * @{
+ */
+
+/**
+ * @brief SENT frame structure
+ */
+struct sent_frame {
+	union {
+		/* ID-Data field for SENT slow frame */
+		struct {
+			uint16_t id;
+			uint16_t data;
+		} slow_frame;
+
+		/* Data field for SENT fast frame */
+		uint32_t data;
+	};
+	uint32_t timestamp;
+	uint8_t crc;
+};
+
+/**
+ * @brief SENT status
+ */
+enum sent_status {
+	SENT_RX_SLOW_FRAME,
+	SENT_RX_FAST_FRAME,
+	SENT_RX_ERR,
```

Comment:
With this design there's no way to distinguish between errors in slow channels and fast channels. I'd imagine it would require different level of actions whether the error occurs in one or the other channel, e.g. fast channel might be a critical impact whereas slow channel might affect only diagnostics.

Reply:
I separate 2 specific error statuses for serial and fast frame

Reply:
Due to there are 2 specific callback for serial and fast frame so we can distinguish between errors